File: umbrella-insurance-react-ui/src/models/users/bots/v1/Bot.ts
import { Analyst } from "../../../people/analysts/v1/Analyst";
import { Unit } from "../../../units/v1/Unit";
import { User } from "../../v1/User";

export class Bot {
    id?: number;
    user?: User;
    botName?: string;
    analyst?: Analyst;
    createdDateTime?: string;
    isDeleted?: boolean;
    isDisabled?: boolean;
    deletedDateTime?: string;
    amountFunded?: number;
    unit?: Unit;//currency

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.user) {
                this.user = new User(obj.user);
            }
            if(obj?.botName) {
                this.botName = obj.botName;
            }
            if(obj?.analyst) {
                this.analyst = new Analyst(obj.analyst);
            }
            if(obj?.createdDateTime) {
                this.createdDateTime = obj.createdDateTime;
            }
            if(typeof obj?.isDeleted === "boolean") {
                this.isDeleted = obj.isDeleted;
            }
            if(typeof obj?.isDisabled === "boolean") {
                this.isDisabled = obj.isDisabled;
            }
            if(obj?.deletedDateTime) {
                this.deletedDateTime = obj.deletedDateTime;
            }
            if(obj?.amountFunded) {
                this.amountFunded = obj.amountFunded;
            }
            if(obj?.unit) {
                this.unit = new Unit(obj.unit);
            }
        }
    }
}

File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Bot.java
package com.umbrella.insurance.core.models.entities;

import com.umbrella.insurance.core.deserializers.TimestampDeserializer;
import com.umbrella.insurance.core.serializers.TimestampSerializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.sql.Timestamp;

@Getter
@Setter
@Entity
@Table(name = "bots", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "bots_unique", columnNames = {"bot_name"})
})
public class Bot {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "bot_id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "user_id")
    private User user;

    @NotNull
    @Column(name = "bot_name", nullable = false, length = Integer.MAX_VALUE)
    private String botName;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "analyst_id")
    private Analyst analyst;

    @JsonSerialize(using = TimestampSerializer.class)
    @JsonDeserialize(using = TimestampDeserializer.class)
    @NotNull
    @Column(name = "created_date_time", nullable = false)
    private Timestamp createdDateTime;

    @Column(name = "is_deleted")
    private Boolean isDeleted;

    @Column(name = "is_disabled")
    private Boolean isDisabled;

    @JsonSerialize(using = TimestampSerializer.class)
    @JsonDeserialize(using = TimestampDeserializer.class)
    @Column(name = "deleted_date_time")
    private Timestamp deletedDateTime;

    @NotNull
    @Column(name = "amount_funded", nullable = false)
    private Double amountFunded;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "unit_id")
    private Unit unit;

}